### ISSUE TITLE
Optic axis norm zero for estimated volume

### ISSUE THREAD

**Issue by gschlafly** [PolarizedLightFieldMicroscopy/forward-model#51]
#### Description
Reconstructions are not possible when starting with an estimated volume containing some optic axis vectors that have norm 0. After the first epoch, the optic axis array contains NaN values.

#### Screenshots
![image](https://github.com/PolarizedLightFieldMicroscopy/forward-model/assets/23206511/1a923d0f-730d-4096-9238-29e0636382ec)

#### Files
When initializing the volume from a file, the optic axis is normalized:
https://github.com/PolarizedLightFieldMicroscopy/forward-model/blob/9c93ee12a1f1bb4c41d5d981519896dc0d9b653d/VolumeRaytraceLFM/birefringence_implementations.py#L103-L104
The Jones matrices calculated when ray tracing contain NaN values:
https://github.com/PolarizedLightFieldMicroscopy/forward-model/blob/9c93ee12a1f1bb4c41d5d981519896dc0d9b653d/VolumeRaytraceLFM/birefringence_implementations.py#L908-L911
The following assertion statements placed in the function above can help debug:
```python
assert not torch.any(torch.isnan(volume_in.optic_axis)), \
    "There is a NaN value in least one optic axis vector."
np_optic_axis = volume_in.optic_axis.detach().numpy()
norms = np.linalg.norm(np_optic_axis, axis=0)
assert not np.any(np.isclose(norms, 0.0)), \
    "One of the optic axis vectors has a norm of zero."
assert not torch.any(torch.isnan(material_JM)), \
    "There is a NaN value in least one Jones matrix."
``` 

#### To Reproduce
Create the initial estimate volume using the following:
```python
h5file_init = 'objects\GUVs\guv4_GT.h5'
volume_estimation = BirefringentVolume.init_from_file(
                                h5file_init,
                                backend=backend,
                                optical_info=optical_info
                                )
``` 
then forward project the volume after one iteration of reconstruction.

#### Tasks
- [x] Identify how the Jones matrices are calculated when the optic axis is a zero vector.
- [x] Check if forward projection of 'objects\GUVs\guv4_GT.h5' works when using a numpy backend.
- [x] Assess whether or not a ground truth object with zero vectors for optic axes can be reasonably valid. 
- [ ] Force all the zero optic axis vector to have a nonzero optic axis, such as `[1,0,0]`

**Comment by gschlafly** [comment]
Optic axis with NaN vectors were created in the optimizer.step() process. I replaced them with random unit vectors in each iteration for the steamlit interface in commit ec7402daa759eaac180164aa23dcd2d10df31589.

**Comment by gschlafly** [comment]
If the optic axis for a voxel is a zero vector, the Jones matric will still be calculated. The projection of the zero vector onto the ray direction vectors will be zero, so `azim = arctan2(0, 0) `. Thus, $\theta = 0, \delta = \Delta n \ell\pi/\lambda$, causing the Jones matrix to be

$$
\begin{pmatrix}
\cos{\delta} + i\sin{\delta} & 0 \\
0 & \cos{\delta} - i\sin{\delta}.
\end{pmatrix}.
$$

If $\Delta n = 0$ too, then the associated Jones matrix is the identity matrix,

$$
\begin{pmatrix}
1 & 0 \\
0 & 1.
\end{pmatrix}.
$$

https://github.com/PolarizedLightFieldMicroscopy/forward-model/blob/ec7402daa759eaac180164aa23dcd2d10df31589/VolumeRaytraceLFM/birefringence_implementations.py#L1133-L1134
